### PR TITLE
Add nested templates support

### PR DESCRIPTION
**Pull request by RyanWhite25**

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
       fail-fast: false
       matrix:
         os: ["macos-latest"]
-        python-version: ["3.9", "3.10", "3.11"]
```
Comment (252afh) [inline .github/workflows/run_tests.yml:10]: not too important but we should be on 3.12 by now, latest is 3.12.7

File: cruft/_commands/link.py [hunk 2]
```diff
@@ -43,6 +44,9 @@ def link(
             project_dir,
             checkout,
         )
+
+        validate_cookiecutter(cookiecutter_template_dir)
```
Comment (252afh) [inline cruft/_commands/link.py:48]: Do the calls to `validate_cookiecutter` need to be in a try:except?

**Review by 252afh — APPROVED** [review]

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (RyanWhite25) [reply .github/workflows/run_tests.yml:10]: Thanks Ell, I should have done this on a separate PR really. I've brought a jira card in to look into upgrading the project to 3.12 and will pick it up once this is merged

File: cruft/_commands/link.py (hunk 2 above)
Comment (RyanWhite25) [reply cruft/_commands/link.py:48]: I've moved this function out of `generate_cookiecutter_context` as it was validating the template too early once nesting was implemented (we want to validate only after the concrete template has been selected). To get around this I've moved the call a layer up the stack but other than that the functionality & error behaviour is the same.

It needs to be a terminating error, and cruft has pretty error handling (see screenshot) so I'm happy for it to live outside of a try/catch. We'd bubble the exception up to the user in any case
image

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (252afh) [reply .github/workflows/run_tests.yml:10]: Sweet, thank you :)

Comment (252afh) [reply .github/workflows/run_tests.yml:10]: I think the biggest breaking changes are 3.10 and 3.13, so if we're migrating across those it's probably a bigger job - though it tends to affect specific libraries that I don't think we have in the cookiecutter

**Pull request merged**